docs/content/src/learn/tutorials/integration-patterns-and-soa/rest-to-soap-service/src/stockquote_service/mock_soap_service.bal

```diff
@@ -85,3 +90,12 @@ function placeOrder() returns xml {
                                 </soapenv:Envelope>`;
     return responsePayload;
 }
+
+function validateSoapAction(string soapAction) returns string {
+    if (soapAction.startsWith("urn:")) {
```

Comment:
What if soapAction is null? Will this throw NPE like Java?

Reply:
I'm just checking it starts with `urn:`, in the related issue soap action has surrounded with \"